timetable/Objects/Constraints/AbstractConstraint.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Xml.Linq;
+using Timetable.timetable.DB;
+using System.Linq;
+
+namespace Timetable.timetable.Objects
+{
+	public abstract class AbstractConstraint
+	{
+		public int weight { get; set; }
+		public XElement constraint { get; set; }
+
+		/// <summary>
+		/// Sets the weight.
+		/// </summary>
+		/// <param name="w">The width.</param>
+		public void SetWeight(int w)
+		{
+			weight = w;
+			constraint.Add(new XElement("Weight_Percentage", weight));
+
+		}
+		/// <summary>
+		/// Sets the element.
+		/// </summary>
+		/// <param name="s">S.</param>
+		public void SetElement(string s)
```

Comment:
Why wrap an additional method around the C# property?  `public XElement constraint { get; set; }` already defines a setter.